### PR TITLE
Pruning cell_data in mesh.prune()

### PR DESCRIPTION
**Pull request by uvilla**
Resolve issue #789

### CODE REVIEW

File: meshio/_mesh.py [hunk 1]
```diff
@@ -74,13 +74,13 @@ def prune(self):
 
         new_cells = []
         new_cell_data = {}
-        for c in self.cells:
+        for block_counter, c in enumerate(self.cells):
```
Comment (nschloe) [inline meshio/_mesh.py:77]: Better `zip`  it, not `enumerate`.

Comment (uvilla) [reply meshio/_mesh.py:77]: Do you mean

```python
for block_counter, c in zip(range(len(self.cells)), self.cells):
```

It seems to me too verbose

Comment (uvilla) [reply meshio/_mesh.py:77]: And I can't either `zip(self.cells, self.cell_data)`, since `self.cell_data` is a Dictionary.

Comment (nschloe) [reply meshio/_mesh.py:77]: You're right. Let's merge this.

**Pull request merged**